src/main/java/edu/ucsb/cs56/ucsb_courses_search/controllers/TutorAssignmentController.java

```diff
@@ -35,15 +34,31 @@ public TutorAssignmentController(TutorAssignmentRepository tutorAssignmentReposi
     this.courseOfferingRepository = courseOfferingRepository;
   }
 
+
   @PostMapping("/tutorAssignments/add")
   public String add(@RequestParam(name = "cid") long cid, @RequestParam(name = "tid") long tid, Model model) {
     Tutor tutor = tutorRepository.findById(tid)
         .orElseThrow(() -> new IllegalArgumentException("Invalid tutor Id:" + tid));
     CourseOffering courseOffering = courseOfferingRepository.findById(cid)
         .orElseThrow(() -> new IllegalArgumentException("Invalid course offering Id:" + cid));
-    TutorAssignment tutorAssignment = new TutorAssignment(tutor, courseOffering);
-    tutorAssignmentRepository.save(tutorAssignment);
-    model.addAttribute("tutorAssignments", tutorAssignmentRepository.findAll());
+    boolean isAssignedQuarter = false;
+
+    Iterator<TutorAssignment> iter = tutor.getTutorAssignments().iterator();
+    while(iter.hasNext()){
+      if(courseOffering.getQuarter().equals(iter.next().getCourseOffering().getQuarter())){
+        isAssignedQuarter = true;
+      }
+    }
```

Comment:
I think this actually is bugged because it won't check the first tutorAssignment in the list.

Reply:
Fixed this by iterating through the iterable with a for loop